sdk/src/stake_weighted_timestamp.rs

```diff
@@ -36,6 +67,48 @@ pub fn calculate_stake_weighted_timestamp(
     }
 }
 
+fn calculate_bounded_stake_weighted_timestamp(
```

Comment:
@sconybeare - care to review this function?

Reply:
Actually, hold off a bit longer. I'm going to rework this tomorrow based on our subsequent discussion

Reply:
@sconybeare, here is the updated bounding function